udata/features/territories/api.py

```diff
@@ -19,14 +20,21 @@ class SuggestTerritoriesAPI(API):
     @api.doc(id='suggest_territory', parser=suggest_parser)
     def get(self):
         args = suggest_parser.parse_args()
-        territories = check_for_territories(args['q'].decode('utf-8'))
+        towns = check_for_towns(args['q'].decode('utf-8'))
+        counties = check_for_counties(args['q'].decode('utf-8'))
+        regions = check_for_regions(args['q'].decode('utf-8'))
+        territories = list(regions) + list(counties) + list(towns)
         if args['size']:
             territories = territories[:args['size']]
+        # Python sort given that we aggregate two distinct querysets.
+        territories = sorted(territories,
+                             key=lambda t: (t.population, t.area),
+                             reverse=True)
         return [{
             'id': territory.id,
             'title': territory.name,
-            'county': territory.county.name,
+            'county': territory.county and territory.county.name or None,
+            'region': territory.region and territory.region.name or None,
```

Comment:
Payload will quickly become heavy when we add level. Can't we just support a single main parent ?